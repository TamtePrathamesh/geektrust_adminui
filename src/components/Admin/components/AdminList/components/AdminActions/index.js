import React from 'react';
import PropTypes from 'prop-types';
import DeleteIcon from '../../../../../images/delete.png';
import EditIcon from '../../../../../images/editing.png';

import { WithBoundary } from '../../../../../common';

import Style from './style.module.css';

const AdminActions = ({ onDelete = () => null, onEditMode = () => null, id = '' }) => {
  return (
    <>
      <div className={Style.actionItems}>
        <img
          data-testid="editRow"
          src={EditIcon}
          alt="cant display"
          className={Style.actionIcons}
          onClick={() => onEditMode(true)}
        />
        <img
          data-testid="deleteRow"
          src={DeleteIcon}
          alt="cant display"
          className={Style.actionIcons}
          onClick={() => onDelete(id)}
        />
      </div>
    </>
  );
};

AdminActions.propTypes = {
  onDelete: PropTypes.func,
  onEditMode: PropTypes.func,
  id: PropTypes.string,
};

export default WithBoundary(React.memo(AdminActions));
