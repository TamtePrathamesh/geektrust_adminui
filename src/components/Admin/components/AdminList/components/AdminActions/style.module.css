.actionItems {
    display: flex;
    align-items: center;
    width: 100%;
    gap: 15px;
}

.actionIcons {
    width: 20px;
    height: 20px;
    cursor: pointer;
}