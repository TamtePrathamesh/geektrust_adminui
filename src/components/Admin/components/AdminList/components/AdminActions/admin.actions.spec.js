import React from 'react';
import { screen, render } from '@testing-library/react';
import user from '@testing-library/user-event';
import AdminActions from '.';

describe('AdminActions', () => {
  it('should match snapshot', () => {
    const onDelete = jest.fn();
    const onEditMode = jest.fn();
    expect(render(<AdminActions onDelete={onDelete} onEditMode={onEditMode} />).container).toMatchSnapshot();
  });

  it('should call functions when buttons are clicked', () => {
    const onDelete = jest.fn();
    const onEditMode = jest.fn();
    render(<AdminActions onDelete={onDelete} onEditMode={onEditMode} />);

    expect(onEditMode).toHaveBeenCalledTimes(0);
    user.click(screen.queryByTestId('editRow'));
    expect(onEditMode).toHaveBeenCalledTimes(1);

    expect(onDelete).toHaveBeenCalledTimes(0);
    user.click(screen.queryByTestId('deleteRow'));
    expect(onDelete).toHaveBeenCalledTimes(1);
  });
});
