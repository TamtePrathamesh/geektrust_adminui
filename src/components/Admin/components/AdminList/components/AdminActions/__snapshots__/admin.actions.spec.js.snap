// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminActions should match snapshot 1`] = `
<div>
  <div
    class="actionItems"
  >
    <img
      alt="cant display"
      class="actionIcons"
      data-testid="editRow"
      src=""
    />
    <img
      alt="cant display"
      class="actionIcons"
      data-testid="deleteRow"
      src=""
    />
  </div>
</div>
`;
